src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,168 @@
+package duke.parser;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ErrorCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * This class contains a function to parse user input.
+ */
+public class Parser {
+
+    /**
+     * Returns the appropriate Command based on the given user input.
+     *
+     * @param input the user input.
+     * @return a Command that should be executed in response to the user input.
+     */
+    public static Command parse(String input) {
+        input = input.strip();
+        String[] split = input.split("\\s+", 2); // guranteed to contain at least ""
```

Comment:
Maybe a more representative name could be used to represent the splited string Array? eg splitStringParts